logging/syslogFormat.go

```diff
@@ -0,0 +1,367 @@
+package logging
+
+import (
+	"fmt"
+	"os"
+	"path"
+	"strings"
+	"time"
+
+	"go.uber.org/zap"
+	"go.uber.org/zap/buffer"
+	"go.uber.org/zap/zapcore"
+)
+
+var (
+	_pool = buffer.NewPool()
+	// Get retrieves a buffer from the pool, creating one if necessary.
+	Get = _pool.Get
+
+	_ zapcore.Encoder = &syslogEncoder{}
+	_                 = zapcore.NewJSONEncoder(zap.NewProductionEncoderConfig()).(genericEncoder)
+)
+
+// Priority maps to the syslog priority levels
+type Priority int
+
+const (
+	version         = 1
+	severityMask    = 0x07
+	facilityMask    = 0xf8
+	nilValue        = "-"
+	timestampFormat = "2006-01-02T15:04:05-07:00"
+	maxHostnameLen  = 255
+	maxAppNameLen   = 48
+
+	NonTransparentFraming Framing = iota
```

Comment:
what does iota mean? SORRY for all the questions this is amazing

Reply:
Iota in Go is used to represent constant increasing sequences

Reply:
Iota in Go is used to represent constant increasing sequences